ethereum/contracts/State.sol

```diff
@@ -41,6 +41,9 @@ contract Storage {
         mapping(address => bool) initializedImplementations;
 
         uint256 messageFee;
+
+        // EIP-155 Chain ID
+        uint256 evmChainId;
```

Comment:
This could go into the provider struct

Reply:
Wouldn't that shift the storage variables declared after provider down a slot and cause issues or am I missing something?